### PR TITLE
[SYCL] Load pi_win_proxy_loader.dll only from trusted directories

### PR DESCRIPTION
**Pull request by againull**
* Embed manifest file into sycl.dll which declares pi_win_proxy_loader.dll
as sycl.dll's private dll file. This will guarantee that pi_win_proxy_loader.dll
is always loaded from the same directory as sycl.dll.
* Exclude current directory (unsafe) from search list for dependencies of sycl.dll,
  sycl-ls, pi_win_proxy_loader using DEPENDENTLOADFLAG linker option.
* When SYCL RT is looking for preloaded plugins don't use path string comparison because
it results in case-sensitivity issues, instead compare parent directory path for equivalency. 

Note. DEPENDENTLOADFLAG can be used to the list of directories for searching. 
Flags can be found here: https://learn.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-loadlibraryexa.

### CODE REVIEW

**Review by cperkinsintel — APPROVED** [review]
sweet

File: sycl/pi_win_proxy_loader/pi_win_proxy_loader.cpp [hunk 1]
```diff
@@ -165,8 +166,25 @@ __declspec(dllexport) void *getPreloadedPlugin(
...
+  std::error_code ec;
+  if (!dllMap.empty() &&
+      std::filesystem::equivalent((dllMap.begin())->first.parent_path(),
+                                  PluginPath.parent_path(), ec)) {
+    // Now we can search only by filename. Result might be nullptr (not found),
+    // which is perfectly valid.
+    match =
+        std::find_if(dllMap.begin(), dllMap.end(),
+                     [&](const std::pair<std::filesystem::path, void *> &v) {
+                       std::error_code ec;
```
Comment (cperkinsintel) [inline sycl/pi_win_proxy_loader/pi_win_proxy_loader.cpp:183]: is `ec` unused?

Comment (againull) [reply sycl/pi_win_proxy_loader/pi_win_proxy_loader.cpp:183]: It is used only to pass it std::filesystem::equivalent - this is the version of std::filesystem::equivalent which doesn't throw exception and returns false if either of directories not equivalent or not found.

Comment (againull) [reply sycl/pi_win_proxy_loader/pi_win_proxy_loader.cpp]: Indeed, removed, thanks!

**Comment by cperkinsintel** [comment]
The tests passed on Windows. Failure seems to be from cleanup.

**Pull request merged**